sdk/core/core/src/az_credential_token_private.h

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef _az_CREDENTIAL_TOKEN_PRIVATE_H
+#define _az_CREDENTIAL_TOKEN_PRIVATE_H
+
+#include <az_credentials.h>
+#include <az_result.h>
+
+#include <stdbool.h>
+
+#include <_az_cfg_prefix.h>
+
+AZ_NODISCARD az_result _az_credential_token_init(_az_credential_token* out_credential_token);
```

Comment:
nit: I understand these are private headers, but consider adding docs.

Reply:
At one point, let's do it all the way across our docs, review everything, do the renames for parameter prefies, replace `@return` with `@retval`s.

Reply:
(resolving, I updated #496 with the idea of documenting private headers)